webapp/app/js/models/resultset.js

```diff
@@ -140,59 +100,44 @@ treeherder.factory('thResultSets', [
                 thUrl.getProjectUrl("/resultset/", repoName),
                 {
                     params: params,
-                    transformResponse:resultSetResponseTransformer
+                    // transformResponse:resultSetResponseTransformer
                 }
             );
         },
         get: function(uri) {
             return $http.get(thServiceDomain + uri, {params: {format: "json"}});
         },
-        getResultSetJobs: function(resultSets, repoName, exclusion_state){
-
-            var uri = '1/get_resultset_jobs/';
-            var fullUrl = thUrl.getProjectUrl("/resultset/", repoName) + uri;
+        getResultSetJobsUpdates: function(resultSetIdList, repoName, exclusionProfile, lastModified){
+            var params = {
+                result_set_id__in: resultSetIdList.join(","),
+                full: "true",
+                count: 5000,
+                last_modified__gt: lastModified
+            };
+            if(exclusionProfile){
+                params.exclusionProfile = exclusionProfile;
+            }
+            return ThJobModel.get_list(repoName, params);
+        },
 
+        getResultSetJobs: function(resultSets, repoName, exclusionProfile){
+            var jobsPromiseList = []
             _.each(
                 resultSets.results,
                 function(rs, index){
                     var params = {
```

Comment:
should this also contain the param ``return_type=list`` to take advantage of that optimization?